File: test/comparison/before.go
package main

import "fmt"

type User struct {
	Name   string
	Email  string
	Role   string
	Active bool
}

const MAX_USERS = 100
const DEFAULT_ROLE = "viewer"

func validateEmail(email string) bool {
	if len(email) == 0 {
		return false
	}
	return true
}

func createUser(name string, email string, role string) User {
	if role == "" {
		role = DEFAULT_ROLE
	}
	if !validateEmail(email) {
		panic("invalid email")
	}
	return User{Name: name, Email: email, Role: role, Active: true}
}

func getUserPermissions(role string) []string {
	if role == "viewer" {
		return []string{"read"}
	}
	if role == "editor" {
		return []string{"read", "write"}
	}
	if role == "admin" {
		return []string{"read", "write", "delete", "manage"}
	}
	return []string{}
}

func deleteUser(userID int) bool {
	fmt.Printf("Deleting user %d\n", userID)
	return true
}

func formatUserDisplay(user User) string {
	return fmt.Sprintf("%s <%s>", user.Name, user.Email)
}

File: test/comparison/before.cpp
#include <string>
#include <stdexcept>
#include <cstdio>

const int MAX_USERS = 100;
const std::string DEFAULT_ROLE = "viewer";

bool validate_email(const std::string &email) {
    if (email.find('@') == std::string::npos) {
        return false;
    }
    if (email.find('.') == std::string::npos) {
        return false;
    }
    return true;
}

struct User {
    std::string name;
    std::string email;
    std::string role;
    bool active;
};

User create_user(const std::string &name, const std::string &email, const std::string &role = DEFAULT_ROLE) {
    if (!validate_email(email)) {
        throw std::runtime_error("Invalid email format");
    }

    User user;
    user.name = name;
    user.email = email;
    user.role = role;
    user.active = true;
    return user;
}

std::string get_user_permissions(const std::string &role) {
    if (role == "viewer") {
        return "read";
    }
    if (role == "editor") {
        return "read,write";
    }
    if (role == "admin") {
        return "read,write,delete,manage";
    }
    return "";
}

void delete_user(int user_id) {
    std::printf("Deleting user %d\n", user_id);
}

std::string format_user_display(const User &user) {
    return user.name + " <" + user.email + ">";
}
